pkg/envoy/xds_server.go

```diff
@@ -1592,26 +1592,26 @@ func getDirectionNetworkPolicy(ep endpoint.EndpointUpdater, l4Policy policy.L4Po
 		return SortPortNetworkPolicies(PerPortPolicies)
 	}
 
-	if len(l4Policy) == 0 {
+	if l4Policy == nil || l4Policy.Len() == 0 {
 		return nil
 	}
 
-	PerPortPolicies := make([]*cilium.PortNetworkPolicy, 0, len(l4Policy))
-	for _, l4 := range l4Policy {
+	PerPortPolicies := make([]*cilium.PortNetworkPolicy, 0, l4Policy.Len())
+	l4Policy.ForEach(func(l4 *policy.L4Filter) bool {
 		var protocol envoy_config_core.SocketAddress_Protocol
 		switch l4.Protocol {
 		case api.ProtoTCP:
 			protocol = envoy_config_core.SocketAddress_TCP
 		case api.ProtoUDP, api.ProtoSCTP:
```

Comment:
Could there be scenarios where protocol is unknown?

Reply:
Not yet, but that will be coming. I'll make a note to convert this into a default case, which it should be.